broker-core/src/test/java/io/zeebe/broker/logstreams/processor/BlacklistInstanceTest.java

```diff
@@ -0,0 +1,288 @@
+import io.zeebe.transport.ServerOutput;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.mockito.Mock;
+
+@RunWith(Parameterized.class)
+public class BlacklistInstanceTest {
+
+  @ClassRule public static ZeebeStateRule zeebeStateRule = new ZeebeStateRule();
+  private static final AtomicLong KEY_GENERATOR = new AtomicLong(0);
+
+  @Parameter(0)
+  public ValueType recordValueType;
+
+  @Parameter(1)
+  public Intent recordIntent;
+
+  @Parameter(2)
+  public boolean expectedToBlacklist;
+
+  private DelegatingEventProcessor delegatingEventProcessor;
+
+  @Parameters(name = "{0} {1} should blacklist instance {2}")
+  public static Object[][] parameters() {
+    return new Object[][] {
+      ////////////////////////////////////////
+      ////////////// DEPLOYMENTS /////////////
+      ////////////////////////////////////////
+      {ValueType.DEPLOYMENT, DeploymentIntent.CREATE, false},
+      {ValueType.DEPLOYMENT, DeploymentIntent.CREATED, false},
+      {ValueType.DEPLOYMENT, DeploymentIntent.DISTRIBUTE, false},
+      {ValueType.DEPLOYMENT, DeploymentIntent.DISTRIBUTED, false},
+
+      ////////////////////////////////////////
+      /////////////// EXPORTER ///////////////
+      ////////////////////////////////////////
+      {ValueType.EXPORTER, ExporterIntent.EXPORTED, false},
+
+      ////////////////////////////////////////
+      ////////////// INCIDENTS ///////////////
+      ////////////////////////////////////////
+      {ValueType.INCIDENT, IncidentIntent.CREATE, true},
+      {ValueType.INCIDENT, IncidentIntent.CREATED, true},
+      {ValueType.INCIDENT, IncidentIntent.RESOLVED, true},
+
+      // USER COMMAND
+      {ValueType.INCIDENT, IncidentIntent.RESOLVE, false},
+
+      ////////////////////////////////////////
+      ////////////// JOB BATCH ///////////////
+      ////////////////////////////////////////
+      {ValueType.JOB_BATCH, JobBatchIntent.ACTIVATE, false},
+      {ValueType.JOB_BATCH, JobBatchIntent.ACTIVATED, false},
+
+      ////////////////////////////////////////
+      //////////////// JOBS //////////////////
+      ////////////////////////////////////////
+      {ValueType.JOB, JobIntent.CREATE, true},
+      {ValueType.JOB, JobIntent.CREATED, true},
+      {ValueType.JOB, JobIntent.ACTIVATED, true},
+      {ValueType.JOB, JobIntent.COMPLETED, true},
+      {ValueType.JOB, JobIntent.TIMED_OUT, true},
+      {ValueType.JOB, JobIntent.FAILED, true},
+      {ValueType.JOB, JobIntent.RETRIES_UPDATED, true},
+      {ValueType.JOB, JobIntent.CANCEL, true},
+      {ValueType.JOB, JobIntent.CANCELED, true},
+      {ValueType.JOB, JobIntent.TIME_OUT, false},
```

Comment:
Didn't we end up saying time out should blacklist as well? I think it's better to be conservative here initially, and we can always change them easily later.

Reply:
No the other 2 we talked about (Job.ACTIVATED and Job.CANCEL), but Time_out we thought that we should not blacklist it, because you can then still complete or fail the task (which I would like to have). But I can do it for now more strict.